torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -822,6 +833,29 @@ ProcessGroupNCCL::ProcessGroupNCCL(
   }
 }
 
+void ProcessGroupNCCL::eagerConnectSingleDevice(at::Device device) {
+  std::vector<at::Device> rankDevices = {device};
+  const auto key = getKeyFromDevices(rankDevices);
+  LOG(INFO) << "Eagerly connecting nccl backend with device " << device;
+  getNCCLComm(key, rankDevices, OpType::ALLREDUCE);
+}
+
+void ProcessGroupNCCL::performNocolorSplit(at::Device device) {
+#ifdef NCCL_HAS_COMM_SPLIT
```

Comment:
QQ: for the #else path, do we keep it a no-op on purpose? If so, do you mind adding a log or comment here? Just for clarity of future developers. Thanks!

Reply:
it's a little ugly but yeah it's a no-op.  the other part of this (the ranks in the new group) will just lazily connect like usual. I'll add a comment.